rewrite-maven/src/test/kotlin/org/openrewrite/maven/MavenSettingsTest.kt

```diff
@@ -327,4 +327,145 @@ class MavenSettingsTest {
+                                <name>Private Repo</name>
+                                <url>https://repo.company.net/maven</url>
+                            </repository>
+                        </repositories>
+                    </profile>
+                </profiles>
+            </settings>
+            """.trimIndent().byteInputStream()
+            }, InMemoryExecutionContext())
+
+            assertThat(settings!!.localRepository).isEqualTo("\${custom.location.zz}/maven/local/repository/")
+            assertThat(settings.servers!!.servers.first().username).isEqualTo("\${env.PRIVATE_REPO_USERNAME_ZZ}")
+            assertThat(settings.servers!!.servers.first().password).isEqualTo("\${env.PRIVATE_REPO_PASSWORD_ZZ}")
+        }
+
+        @Test
+        @Disabled("Depends on methods which are unstable in CI build")
+        fun env() {
+            updateEnvMap("REWRITE_TEST_PRIVATE_REPO_USERNAME", "user")
+            updateEnvMap("REWRITE_TEST_PRIVATE_REPO_PASSWORD", "pass")
+            val settings = MavenSettings.parse(Parser.Input(Paths.get("settings.xml")) {//language=xml
+                """
+            <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+                <servers>
+                     <server>
+                         <id>private-repo</id>
+                         <username>${'$'}{env.REWRITE_TEST_PRIVATE_REPO_USERNAME}</username>
+                         <password>${'$'}{env.REWRITE_TEST_PRIVATE_REPO_PASSWORD}</password>
+                    </server>
+                </servers>
+                <localRepository>/tmp/maven/local/repository/</localRepository>
+                <activeProfiles>
+                    <activeProfile>
+                        my-profile
+                    </activeProfile>
+                </activeProfiles>
+                <profiles>
+                    <profile>
+                        <id>my-profile</id>
+                        <repositories>
+                            <repository>
+                                <id>private-repo</id>
+                                <name>Private Repo</name>
+                                <url>https://repo.company.net/maven</url>
+                            </repository>
+                        </repositories>
+                    </profile>
+                </profiles>
+            </settings>
+            """.trimIndent().byteInputStream()
+            }, InMemoryExecutionContext())
+
+            assertThat(settings?.servers!!.servers).hasSize(1)
+            assertThat(settings.servers!!.servers.first().username).isEqualTo("user")
+            assertThat(settings.servers!!.servers.first().password).isEqualTo("pass")
+        }
+
+        /**
+         * Unstqble in CI builds
```

Comment:
```suggestion
         * Unstable in CI builds
```
In what way is this unstable?

Reply:
I believe we are now running our [CI builds on Java 17](https://github.com/openrewrite/gh-automation/blob/main/.github/workflows/ci-gradle.yml#L66)

So reflective access to non-open modules is going to result in an error in Java 17..but not in earlier versions of Java.

Reply:
that would do it -- I've been building on jdk11 on my machine.